joinservice/internal/watcher/validator.go

```diff
@@ -76,11 +81,47 @@ func (u *Updatable) Update() error {
 	}
 	u.log.Debugf("New expected measurements: %+v", cfg.GetMeasurements())
 
-	validator, err := choose.Validator(cfg, u.log)
+	cfgWithCerts, err := u.configWithCerts(cfg)
+	if err != nil {
+		return fmt.Errorf("adding cached certificates: %w", err)
+	}
+
+	validator, err := choose.Validator(cfgWithCerts, u.log)
 	if err != nil {
-		return fmt.Errorf("updating validator: %w", err)
+		return fmt.Errorf("choosing validator: %w", err)
 	}
 	u.Validator = validator
 
 	return nil
 }
+
+// addCachedCerts adds the certificates cached by the validator to the attestation config, if applicable.
+func (u *Updatable) configWithCerts(cfg config.AttestationCfg) (config.AttestationCfg, error) {
+	switch c := cfg.(type) {
+	case *config.AzureSEVSNP:
+		ask, err := u.getCachedAskCert()
+		if err != nil {
+			return nil, fmt.Errorf("getting cached ASK certificate: %w", err)
+		}
+		c.AMDSigningKey = config.Certificate(ask)
+		return c, nil
+	}
+	// TODO(derpsteb): Add AWS SEV-SNP
+
+	return cfg, nil
+}
+
+// getCachedAskCert returns the cached SEV-SNP ASK certificate.
+func (u *Updatable) getCachedAskCert() (x509.Certificate, error) {
+	if u.cachedCerts == nil {
+		return x509.Certificate{}, fmt.Errorf("no cached certs available")
+	}
+	ask, ark := u.cachedCerts.SevSnpCerts()
+	if ask == nil {
+		return x509.Certificate{}, fmt.Errorf("no ASK available")
+	}
+	if ark == nil {
+		return x509.Certificate{}, fmt.Errorf("no ARK available")
+	}
```

Comment:
```suggestion
	ask, _ := u.cachedCerts.SevSnpCerts()
	if ask == nil {
		return x509.Certificate{}, fmt.Errorf("no ASK available")
	}
```

Reply:
Do we only want to cache the ASK certificate completely for now? This would let us remove a lot of the logic inside this function, but also closes the door on keeping the ARK here instead of the Constellation config.

Reply:
Note that this code is from the watcher. If it only needs the ASK (for now), I wouldn't let it fail here.

Regarding "a lot of the logic inside this function", I guess you mean inside certcache? I also thought about this, but wasn't sure. That's why I only commented on the client of that cache (i.e., the watcher validator).

Reply:
Yes, my reply was in regards to the certcache, sorry. I think it would let us spare some code now, but since the general attestation design (i.e. ARK -> ASK certchain) should be somewhat stable, I don't think it would require a lot of maintenance if its tested well enough. But I don't have a strong opinion on removing it either.